servercom/servercom.go

```diff
@@ -0,0 +1,138 @@
+package serverCom
+
+import (
+	"fmt"
+	mm "github.com/mattermost/platform/model"
+)
+/*
+ServerCom acts as a mitigator between the frontend and the mattermost model API.
+*/
+type ServerCom struct {
+	Client mm.Client
+	Team mm.Team
+	Channel *mm.Channel
+}
+
+/*
+Startup accepts the url and login credentials for a user, and returns a new serverCom struct.
+*/
+func Startup(url string, username string, password string) ServerCom {
```

Comment:
I have a couple of recommendations here. Since this method is used to construct `ServerCom` structs, and since all of your methods are defined on `*ServerCom` pointers, you might as well return a pointer here.

On another note, error handling in library functions can be counterintuitive. I quite like [this guy's thoughts](https://dave.cheney.net/2016/04/27/dont-just-check-errors-handle-them-gracefully) on the subject. At the least, I would recommend changing the return types of most of these functions to return an error in addition to their normal return type. Then, instead of printing the error, return it. 
The reason for this is that you don't really want library functions dumping things onto Stdout. That makes it very hard to control what is printed to the screen from the `main()` function of your application. If you return errors to main, then main can decide to print them (or not to) at its discretion. Does that make sense?